completions/7z

```diff
@@ -114,7 +114,7 @@ _7z()
     else
         if [[ ${words[1]} == d ]]; then
             local IFS=$'\n'
-            COMPREPLY=($(compgen -W "$(printf '%s\n' "$($1 l ${words[2]} \
+            COMPREPLY=($(compgen -W "$(printf '%s\n' "$($1 l "${words[2]}" \
```

Comment:
```suggestion
            COMPREPLY=($(compgen -W "$(printf '%s\n' "$("$1" l "${words[2]}" \
```

Maybe this might not be requested by `shellcheck`, but the command names are also subject to word splitting and pathname expansions. Hmm, but there seem to be again many similar cases in the whole tree:

```bash
$ git grep -n '\$(\$' | wc
92
$ git grep -n '\$("\$' | wc
70
```

<details><summary>List of <code>$($...</code></summary>

```bash
$ git grep -n '\$(\$'
completions/2to3:14:                "$($1 --list-fixes 2>/dev/null | command sed -e 1d)" -- "$cur"))
completions/7z:117:            COMPREPLY=($(compgen -W "$(printf '%s\n' "$($1 l "${words[2]}" \
completions/_adb:42:            tmp+=($($1 help 2>&1 | awk '$1 == "adb" { print $2 }'))
completions/_mount:57:        COMPREPLY=($(compgen -W "$($1 | awk '! /^[ \t]*#/ {if ($3 ~ /\//) print $3}')" -- "$cur"))
completions/_rfkill:21:                    COMPREPLY=($(compgen -W "$($1 list | awk -F: \
completions/abook:25:            COMPREPLY=($(compgen -W "$($1 --formats |
completions/abook:30:            COMPREPLY=($(compgen -W "$($1 --formats |
completions/apache2ctl:9:    APWORDS=$($1 2>&1 >/dev/null | awk 'NR<2 { print $3; exit }' |
completions/appdata-validate:13:            COMPREPLY=($(compgen -W "$($1 --help |
completions/aptitude:75:        local opts=" $($1 --help 2>&1 | command sed -e \
completions/aspell:6:    datadir=$($aspell config data-dir 2>/dev/null || echo /usr/lib/aspell)
completions/aspell:14:    COMPREPLY+=($($aspell dicts 2>/dev/null))
completions/aspell:38:            COMPREPLY=($(compgen -W "$($1 modes 2>/dev/null |
completions/aspell:56:            COMPREPLY=($(compgen -W "$($1 filters 2>/dev/null |
completions/brctl:21:                    COMPREPLY=($(compgen -W "$($1 show |
completions/ccache:23:                COMPREPLY=($(compgen -S = -W "$($1 -p 2>/dev/null |
completions/checksec:29:            COMPREPLY=($(compgen -W '$($1 --help 2>/dev/null |
completions/checksec:36:            COMPREPLY=($(compgen -W '$($1 --help 2>/dev/null |
completions/chronyc:6:    args=($(compgen -W "$($1 help 2>/dev/null |
completions/chronyc:44:            COMPREPLY=($(compgen -W "$($1 help 2>/dev/null |
completions/configure:31:        COMPREPLY=($(compgen -W "$($1 --help 2>&1 |
completions/cvsps:13:            COMPREPLY=($(compgen -W "$($1 2>/dev/null |
completions/cvsps:18:            COMPREPLY=($(compgen -W "$($1 2>/dev/null |
completions/cvsps:23:            COMPREPLY=($(compgen -W "$($1 2>/dev/null |
completions/cvsps:28:            COMPREPLY=($(compgen -W "$($1 2>/dev/null |
completions/dmypy:39:        local cmds=$($1 --help 2>&1 |
completions/gcc:12:            local cc=$($1 -print-prog-name=cc1 2>/dev/null)
completions/gcc:14:            COMPREPLY=($(compgen -W "$($cc --help 2>/dev/null | tr '\t' ' ' |
completions/gcc:54:        local flags=$($1 --completion="$argument" | cut -c $prefix_length-)
completions/getent:45:            COMPREPLY=($(compgen -W "$($1 "$db" |
completions/getent:50:            COMPREPLY=($(compgen -W "$($1 "$db" | cut -d: -f1)" -- "$cur"))
completions/gpg:20:            COMPREPLY=($(compgen -W "$($1 --list-keys 2>/dev/null |
completions/gpg:29:            COMPREPLY=($(compgen -W "$($1 --list-secret-keys 2>/dev/null |
completions/gpg:34:            COMPREPLY=($(compgen -W "$($1 --list-keys 2>/dev/null |
completions/gpg:46:        COMPREPLY=($(compgen -W '$($1 --dump-options)' -- "$cur"))
completions/gpg2:21:            COMPREPLY=($(compgen -W "$($1 --list-keys 2>/dev/null |
completions/gpg2:28:            COMPREPLY=($(compgen -W "$($1 --list-keys 2>/dev/null |
completions/gpg2:40:        COMPREPLY=($(compgen -W '$($1 --dump-options)' -- "$cur"))
completions/gphoto2:26:            COMPREPLY=($(compgen -W "$($1 --list-ports 2>/dev/null |
completions/gphoto2:33:            COMPREPLY=($(compgen -W "$($1 --list-cameras 2>/dev/null |
completions/gssdp-discover:18:            local types=$($1 --help 2>&1 |
completions/iconv:5:    COMPREPLY+=($(compgen -X ... -W '$(${1:-iconv} -l | \
completions/influx:17:            local args=$($1 --help 2>&1 | awk "\$1 == \"$prev\" { print \$2 }")
completions/inotifywait:8:    COMPREPLY+=($(compgen -W "$($1 --help 2>/dev/null |
completions/ipmitool:5:    COMPREPLY=($(compgen -W "$($1 "$2" 2>&1 |
completions/ipmitool:26:            COMPREPLY=($(compgen -W "$($1 -h 2>&1 |
completions/ipmitool:54:            COMPREPLY=($(compgen -W "$($1 -o list 2>&1 |
completions/ipv6calc:14:            COMPREPLY=($(compgen -W "$($1 "$prev" -h 2>&1 |
completions/lspci:21:            COMPREPLY+=($(compgen -W '$($1 -A help | command grep -vF :)' \
completions/makepkg:18:            '$($1 | command sed -e "s/^options://" | _parse_help -)' \
completions/mcrypt:15:            COMPREPLY=($(compgen -W '$($1 --list-keymodes 2>/dev/null )' \
completions/mcrypt:20:            COMPREPLY=($(compgen -W "$($1 --list 2>/dev/null | cut -d: -f2-)" \
completions/mcrypt:25:            COMPREPLY=($(compgen -W "$($1 --list 2>/dev/null |
completions/mcrypt:30:            COMPREPLY=($(compgen -W '$($1 --list-hash 2>/dev/null | \
completions/medusa:18:            COMPREPLY=($(compgen -W "$($1 -d | awk '/^ +\+/ {print $2}' |
completions/mplayer:6:    COMPREPLY=($(compgen -W "$($1 -noconfig all "$2" help 2>/dev/null |
completions/mplayer:276:            COMPREPLY=($(compgen -W '$($cmd -noconfig all -list-options 2>/dev/null | \
completions/msynctool:10:            COMPREPLY=($(compgen -W "$($1 --showgroup \
completions/msynctool:16:            COMPREPLY=($(compgen -W '$($1 --listplugins \
completions/msynctool:24:            COMPREPLY=($(compgen -W '$($1 --listgroups \
completions/msynctool:29:            COMPREPLY=($(compgen -W '$($1 --listobjects \
completions/mutt:116:    querycmd="$($muttcmd -Q query_command 2>/dev/null | command sed -e 's|^query_command=\"\(.*\)\"$|\1|' -e 's|%s|'"$cur"'|')"
completions/mutt:121:        queryresults=($($querycmd |
completions/mutt:135:        folder="$($muttcmd -F "$muttrc" -Q folder 2>/dev/null | command sed -e 's|^folder=\"\(.*\)\"$|\1|')"
completions/mutt:145:        spoolfile="$($muttcmd -F "$muttrc" -Q spoolfile 2>/dev/null |
completions/openssl:108:                COMPREPLY=($(IFS=: compgen -W "$($1 ciphers)" -- "$cur"))
completions/perl:6:        "$(${2:-perl} "${BASH_SOURCE[0]%/*}/../helpers/perl" "$1" "$cur")" \
completions/perl:54:                        '$($1 -MConfig -e "print join \"\\n\",
completions/pgrep:38:                $(IFS="$IFS," compgen -W '$($1 --help 2>&1 |
completions/pkg-config:38:        COMPREPLY=($(compgen -W "$($1 --list-all \
completions/puppet:30:        cert_list=$($puppetca --list)
completions/pydoc:36:        '$($1 keywords topics | command sed -e /^Here/d)' -- "$cur"))
completions/python:6:        "$(${1:-python} "${BASH_SOURCE[0]%/*}/../helpers/python" "$cur" \
completions/qemu:30:            COMPREPLY=($(compgen -W "$($1 -soundhw help | awk \
completions/qemu:35:            COMPREPLY=($(compgen -W "$($1 "$prev" help | awk \
completions/qemu:40:            COMPREPLY=($(compgen -W "$($1 -cpu help | awk '{print $2}')" \
completions/qemu:85:            COMPREPLY=($(compgen -W "$($1 -watchdog help 2>&1 |
completions/rpm:17:        COMPREPLY=($(${1:-rpm} -qa --nodigest --nosignature \
completions/rpm:27:    COMPREPLY=($(compgen -W "$(${1:-rpm} -qa --nodigest --nosignature \
completions/rpm:34:    COMPREPLY=($(compgen -W "$(${1:-rpm} --showrc | command sed -ne \
completions/rpm:41:    COMPREPLY=($(compgen -W "$(${1:-rpm} --showrc | command sed -ne \
completions/rpm:108:                COMPREPLY=($(compgen -W "$($1 -qa --nodigest --nosignature \
completions/rpm:266:            local cfgdir=$($rpm --eval '%{_rpmconfigdir}' 2>/dev/null)
completions/sitecopy:44:        COMPREPLY=($(compgen -W "$($1 -v |
completions/svk:193:                    COMPREPLY=($(compgen -W "$($1 mirror --list \
completions/svk:202:                    COMPREPLY=($(compgen -W "$($1 list "$path" 2>/dev/null |
completions/synclient:17:        COMPREPLY=($(compgen -S = -W '$($1 -l 2>/dev/null | \
completions/sysbench:80:            COMPREPLY=($(compgen -W "$($1 --test=oltp help 2>/dev/null |
completions/tar:722:    line="$($cmd --version 2>/dev/null)"
completions/valgrind:76:            local value=$($1 --help-debug ${tool-} 2>/dev/null |
completions/wodim:47:                COMPREPLY=($(compgen -W "$($1 driver=help 2>&1 |
completions/xfreerdp:54:        local help="$($1 --help)"
```
</details>

Reply:
Yeah, I'm pretty sure there are still even after those `$1` fixes quite a few in need of treatment. But I'll address those you flagged now while we're at it.

Reply:
Done in cb7d8c07b13ea45d6bf8d923baa6664da8ea8077, let's see if I managed to break something caught by CI.